### PR TITLE
Fixes weekly repetition not generating any sessions

### LINKED ISSUE #18: No sessions generating when repeating weekly
## Symptoms

No sessions are generated when repeating weekly, and the rule's duration is more than 1 day.

## Steps to Reproduce

**In EDD Bookings:**
1. Create a service.
2. Set a session length of 1 day.
3. Create a rule of length 4 days.
4. Set the rule to repeat every 1 week.
5. Set the rule to stop repeating after some weeks.
6. Save the rule and the service.

**Using the EDD Bookings wizard:**
7. Visit the front-end wizard and select the service.
8. Observe the symptom - no sessions are available.

**Using the REST API:**
7. Send a GET request to `/wp-json/eddbk/v1/sessions?service=S` where `S` is the service ID
8. Observe the symptom - no sessions in response and `count` is `0`.

## Possible Causes

Since the day-of-the-week options are not shown for rules that are longer than a day, the server will receive no days of the week, and will not generate sessions.

## Recommended Solutions

Detect rules that are longer than a day and do not apply day-of-the-week processing for such rules.

### PR DESCRIPTION
**Pull request by mecha**
Fixes #18

### CODE REVIEW

File: src/Module/SessionRuleFactory.php [hunk 1]
```diff
@@ -58,6 +58,13 @@ class SessionRuleFactory implements FactoryInterface
     /* @since [*next-version*] */
     use StringTranslatingTrait;
 
+    /**
+     * The duration up till which the weekly day-of-the-week repetition processing applies.
+     *
+     * @since [*next-version*]
+     */
+    const WEEKLY_REPEAT_DOTW_DURATION_THRESHOLD = 86400;
```
Comment (XedinUnknown) [inline src/Module/SessionRuleFactory.php:66]: Could you convert the value to a multiplication expression? This would add clarity WRT how long this is.

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

File: src/Module/SessionRuleFactory.php (hunk 1 above)
Comment (mecha) [reply src/Module/SessionRuleFactory.php:66]: Not allowed prior to 5.6

Comment (mecha) [reply src/Module/SessionRuleFactory.php:66]: I can document it instead.

**Review by XedinUnknown — APPROVED** [review]

File: src/Module/SessionRuleFactory.php (hunk 1 above)
Comment (XedinUnknown) [reply src/Module/SessionRuleFactory.php:66]: Too late ;P

**Pull request merged**